sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -61,8 +68,15 @@ public static void setIOFactory(String scheme, IOChannelFactory factory) {
    * to provide, e.g., credentials for GCS.
    */
   public static void registerStandardIOFactories(PipelineOptions options) {
-    setIOFactory("gs", new GcsIOChannelFactory(options.as(GcsOptions.class)));
-    setIOFactory("file", new FileIOChannelFactory());
+    Set<IOChannelFactoryFactoryRegistrar> registrars =
+        Sets.newTreeSet(ReflectHelpers.ObjectsClassComparator.INSTANCE);
+    registrars.addAll(Lists.newArrayList(
+        ServiceLoader.load(IOChannelFactoryFactoryRegistrar.class, CLASS_LOADER)));
+    for (IOChannelFactoryFactoryRegistrar registrar : registrars) {
```

Comment:
We should make sure that multiple factories for the same scheme is an error with an appropriate error message.

We need a test to show that the registration is occurring similar to whats in PipelineOptionsFactoryTest